### ISSUE TITLE
Variable Product - if select `0` as attribute value, Customer gets an error message `Attribute-xxx is a required field`

### ISSUE THREAD

**Issue by kosiew** [woocommerce/woocommerce#27618]
Please provide us with the information requested in this bug report. Without these details, we won't be able to fully evaluate this issue. 
Bug reports lacking detail, or for any other reason than to report a bug, may be closed without action.

**Prerequisites (mark completed items with an [x]):**
- [ ] I have have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**

When User chooses attribute value `0` for a variable product, he gets an error message that the Attribute is a required value.

This bug was introduced in 4.5.
Tested the same with 4.4.1 without issues.

**Expected behavior**
I should be able to choose any of the attribute values including `0`.

**Actual behavior**
When `0` is chosen, I get error message that the attribute is a required field.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1.  set up this variable product:
![Screenshot](https://d.pr/i/BuA0Cq+)
*if image does not load, please click on this link  : https://d.pr/i/BuA0Cq*

and this variation:

![Screenshot](https://d.pr/i/8r2eLf+)
*if image does not load, please click on this link  : https://d.pr/i/8r2eLf*

2. Try to purchase the `0` variation,I get this error

![Screenshot](https://d.pr/i/YFBH5Y+)
*if image does not load, please click on this link  : https://d.pr/i/YFBH5Y*

**Screenshots**

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
We use the [WooCommerce System Status Report](https://docs.woocommerce.com/document/understanding-the-woocommerce-system-status-report/) to help us evaluate the issue. 
Without this report we won't be able to fully evaluate this issue.
<details>
```
The System Status Report is found in your WordPress admin under **WooCommerce > Status**. 
Please select “Get system report”, then “Copy for support”, and then paste it here.
```
</details>

**Comment by juliaamosova** [comment]
Hi @kosiew,

Thank you for taking the time to report this bug, we really appreciate your help. I can reproduce it on my end. 

Documenting my steps below:

I have a variable product set up as described in the report:

![Screen Shot 2020-09-09 at 9 01 54 AM](https://user-images.githubusercontent.com/19143190/92601858-5178ec80-f27b-11ea-82b3-66a38d32a690.png)

![Screen Shot 2020-09-09 at 9 02 07 AM](https://user-images.githubusercontent.com/19143190/92601873-55a50a00-f27b-11ea-8aad-2a8f2c68417a.png)

I can view the product on the site and select `0` value but the error message comes up when I click on the `Add to cart` button:

![Screen Shot 2020-09-09 at 9 02 27 AM](https://user-images.githubusercontent.com/19143190/92601927-6a819d80-f27b-11ea-97bd-0aace333797d.png)

We’re going to solve this bug as soon as possible.

**Comment by juliaamosova** [comment]
Noting that this issue should be resolved via https://github.com/woocommerce/woocommerce/pull/27625 which will be shipped in WC 4.5.1 asap.
